syncd/syncd_hard_reinit.cpp

```diff
@@ -157,10 +157,56 @@ void checkAllIds()
     redisSetVidAndRidMap(g_translated);
 }
 
+void saiRemoveDefaultVlanMembers()
+{
+    SWSS_LOG_ENTER();
+
+    // after asic init, all ports are vlan 1 members
+    // we will remove them to not complicate reinit
+    // if user want to add vlan 1, then it needs to be
+    // done explicitly.
+
+    const auto& portList = saiGetPortList();
+
+    std::vector<sai_object_id_t> vlanMemberList;
+
+    vlanMemberList.resize(portList.size());
+
+    sai_attribute_t attr;
+
+    attr.id = SAI_VLAN_ATTR_MEMBER_LIST;
+
+    attr.value.objlist.count = vlanMemberList.size();
+    attr.value.objlist.list = vlanMemberList.data();
+
+    sai_status_t status = sai_vlan_api->get_vlan_attribute(DEFAULT_VLAN_NUMBER, 1, &attr);
```

Comment:
then you need resize the vlanMemberList otherwise, in line 191 it enumernate many members.